logstash-core/src/main/java/org/logstash/instrument/metrics/gauge/LazyDelegatingGauge.java

```diff
@@ -89,6 +90,10 @@ private synchronized void wakeMetric(Object value) {
                 lazyMetric = new BooleanGauge(nameSpaces, key, (Boolean) value);
             } else if (value instanceof RubyHash) {
                 lazyMetric = new RubyHashGauge(nameSpaces, key, (RubyHash) value);
+            } else if (value instanceof RubyTimestamp) {
+                lazyMetric = new RubyTimeStampGauge(nameSpaces, key, ((RubyTimestamp) value));
```

Comment:
I logged https://github.com/elastic/logstash/issues/7746 to address the deprecation as a separate issue. Until we remove / move the underlying metrics that uses this, users logs will get spammed with warnings with no re-course. Once the core/known RubyTimeStampGauge usage is moved/removed, the I will add the deprecation warning to the log.